crates/iota-node/src/lib.rs

```diff
@@ -584,6 +584,10 @@ impl IotaNode {
             signature_verifier_metrics,
             &config.expensive_safety_check_config,
             ChainIdentifier::from(*genesis.checkpoint().digest()),
+            checkpoint_store
+                .get_highest_executed_checkpoint_seq_number()
+                .expect("checkpoint store read cannot fail")
```

Comment:
```suggestion
                .get_highest_executed_checkpoint_seq_number()?
```
Why not error propagation here?

Reply:
I'll keep this now, and further clean up in https://github.com/iotaledger/iota/issues/7909